pkg/storage/gc/retention/candidate/candidate.go

```diff
@@ -0,0 +1,40 @@
+package candidate
```

Comment:
Why make so many subdirs for this? Just add all the files under the pkg/storage/retention.go? instead? It is a storage feature right?

Reply:
the rules implements an interface, that's why I made each one a package.
candidate package is both used by gc package and by rules (avoid circular import)
same with regex.

yes, currently retention is implemented inside gc, because some functionalities were already in gc, like deleteReferrers, deleteUntagged.

Reply:
That is why we have pkg/storage/types.go etc

All interfaces go there.

Reply:
There are multiple implementations each in a package, what does pkg/storage/types.go have to do with this topic?